rabbitmq/check.py

```diff
@@ -276,14 +277,27 @@ def get_stats(self, instance, base_url, object_type, max_detailed, filters, cust
         filters: explicit or regexes filters of specified queues or nodes (specified in the yaml file)
         """
         instance_proxy = self.get_instance_proxy(instance, base_url)
-        data = self._get_data(urlparse.urljoin(base_url, object_type), auth=auth,
-                              ssl_verify=ssl_verify, proxies=instance_proxy)
-
         # Make a copy of this list as we will remove items from it at each
         # iteration
         explicit_filters = list(filters['explicit'])
         regex_filters = filters['regexes']
 
+        data = []
+
+        # only do this if vhosts were specified,
+        # otherwise it'll just be making more queries for the same data
+        if self._limit_vhosts(instance) and object_type == QUEUE_TYPE:
```

Comment:
We already have `vhosts` passed as param, I think this could be:
```python
if vhosts and object_type == QUEUE_TYPE:
```

Reply:
We only want to grab from every vhost endpoint if vhosts are limited in the instance yaml. The vhosts param is populated regardless of this, either with what they limited in there, or with the info of every vhost, queried from the API. If we're not limiting the vhosts, I think we should use the global endpoint instead of making an API call that mimics the global endpoint.